testgrid.tahoe-lafs.org/repo-tahoe-lafs-master.nix

```diff
@@ -0,0 +1,9 @@
+let
+  pkgs = import <nixpkgs> {};
+in
+  pkgs.fetchFromGitHub {
+    owner = "tahoe-lafs";
+    repo = "tahoe-lafs";
+    rev = "49df402f0762b34c88b01d183b9d217da117cc79";
```

Comment:
File-name says "master" .. but I guess it's not possible to literally say "master" because that's not repeat-able in Nix-land?

Reply:
What do we need in order to have continuous deployment? (Or, what I believe to be continuous deployment... as in, can the grid always run the latest of what's on tahoe-lafs master branch?)

Reply:
> File-name says "master" .. but I guess it's not possible to literally say "master" because that's not repeat-able in Nix-land?

Nix will let us say `master@HEAD` if we want to.  It is true that this makes the build non-repeatable so that's why I haven't done it here.

> What do we need in order to have continuous deployment? (Or, what I believe to be continuous deployment... as in, can the grid always run the latest of what's on tahoe-lafs master branch?)

There are some options for this.  We replace this expression with `builtins.fetchTarball "https://api.github.com/repos/tahoe-lafs/tahoe-lafs/tarball/master"`.  That gives us the kind of non-repeatable build I described above.  Every time we run a deployment we'll get whatever `master@HEAD` resolves so.  Then continuous deployment is just kicking off a deploy every time Tahoe-LAFS master@HEAD moves.

Another option would be a bit less continuous - bump the revision in this expression every time Tahoe-LAFS master@HEAD moves.  This could be done manually or automatically in a dependabot-style workflow (some bot watches Tahoe-LAFS master@HEAD and sends us a PR that includes the new revision whenever it moves).

Whether this is manual or automated, this gives us a little more control over when the deployment is updated and keeps the build repeatable (since every infrastructure main@HEAD revision points at a specific Tahoe-LAFS revision).  If it isn't automated then it's kind of tedious.  Also I haven't tried setting up this kind of automation before (and it would be nice to avoid building more GitHub-lock-in into our workflow at this point...)